test/functional/feature_assumeutxo.py

```diff
@@ -12,6 +12,9 @@
 ## Possible test improvements
 
 - TODO: test submitting a transaction and verifying it appears in mempool
+- TODO: the "assumed" field for a confirmed wallet transaction should be
+        present during background sync, start false and become true as
+        sync progresses
```

Comment:
Interesting that you're getting that error when loading from a backup, but not when loading a regular wallet.

Reply:
> I think wallet tests should be separate from consensus tests.

That makes sense.

Reply:
Added a wallet test file here: 59796c3926cb146229a2b3e942a5f6994f27d72a

It made me realise we should probably disallow loading a wallet with a birth date below the assume valid block. Though for manual testing it's handy.

Reply:
I'm still trying to wrap my head around the logic in `CWallet::AttachChain` which is what triggers the error when loading a backup, but somehow doesn't trigger it when loading an existing wallet.

Reply:
Oh I see, the test creates the backup _before_ the snapshot height.

Reply:
As for the behavior that this PR introduces, ideally there would a `stopatheight` RPC, but the test could also - as it does now - use `-stopatheight` and restart the node. Since it won't automatically connect to the other test nodes, we can inspect the result of `getwallettransactions` and see if `assumed` is set correctly.

I'll work on that and make it a separate commit, so it's a bit easier to extract the test scaffold if this PR doesn't make it.